app/src/main/java/com/iti/android/team1/ebuy/ui/product_details_screen/viewmodel/ProductDetailsVMFactory.kt

```diff
@@ -8,6 +8,8 @@ class ProductDetailsVMFactory(private val myRepo: IRepository) : ViewModelProvid
     override fun <T : ViewModel> create(modelClass: Class<T>): T {
         if (modelClass.isAssignableFrom(ProductsDetailsViewModel::class.java)) {
             return ProductsDetailsViewModel(myRepo) as T
+        } else if (modelClass.isAssignableFrom(AddToCartViewModel::class.java)) {
```

Comment:
The class should have single responsibility? should we create one factory that returns all the viewmodels ?

Reply:
I use the same Factory since the dialogue will appear on the same screen. But if it's in another Factory, I'll absolutely do it.

Reply:
so should we use this factory for all the other view models?

Reply:
Ok